docs/guides/spmd_api.md

```diff
@@ -0,0 +1,429 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.13.8
+  kernelspec:
+    display_name: Python 3 (ipykernel)
+    language: python
+    name: python3
+---
+
+<!-- #region id="2-JYevEklHfB" -->
+# Scale up Flax modules with `pjit`
+
+In this guide, you will learn how to use JAX's `pjit` and `flax.linen.spmd` to scale up your Flax modules on multiple devices and hosts.
+
+Modern JAX provides [`jax.experimental.pjit`](https://jax.readthedocs.io/en/latest/jax.experimental.pjit.html) as a way to automatically compile and scale up JAX computations using the [SPMD](https://jax.readthedocs.io/en/latest/glossary.html?highlight=spmd#term-SPMD) model. You can think of `pjit` as [`jit`](https://jax.readthedocs.io/en/latest/jax-101/02-jitting.html) where one explicitly specificies how the input and output data should be partitioned across devices. See the [JAX guide](https://jax.readthedocs.io/en/latest/jax-101/08-pjit.html) for more information on pjit.
+
+Flax provides an interface for you to specify partitions for your module parameters when defining your module, and later use it for `pjit` compilation. You can also use logical axis annotations to decouple your model code and partition plan, in order to customize and experiment with different partition layouts more easily.
+<!-- #endregion -->
+
+<!-- #region id="eTGIfmDclxJ2" -->
+# Setup
+
+Install Flax from head and make imports as necessary. 
+
+Note that this Colab uses `--xla_force_host_platform_device_count=8` to emulate multiple devices on a CPU environment. You may also choose to run on a multi-device TPU environment following [this guide](https://jax.readthedocs.io/en/latest/jax-101/08-pjit.html#setup), in which case you should ignore the `os.environ` cell.
+<!-- #endregion -->
+
+```python id="e3d005e672d6"
+# TODO: replace this with `pip3 install flax` when 0.6.4 release rolls out.
```

Comment:
Since this is a public doc, it's probably better to not expose our names. I can open an issue to remind myself of that.

Reply:
fyi agreed with @IvyZX 

```
# Once Flax v0.6.4 is out, replace this with `pip3 install flax`.
```

This way, the docs can be more or less up-to-date after 0.6.4.

Reply:
You could use Github username (which is already known) ... Or create an issue and assign it to yourself.

The main motivation here is to avoid accumulating TODOs in public docs.